1.14.4/x86_64-bionic/Dockerfile

```diff
@@ -1,7 +1,52 @@
-FROM ubuntu:bionic
+FROM ubuntu:bionic AS verify
 
-# To improve : static hash make dynamic build of versions impossible.
-ARG VERSION=1.14.4
+WORKDIR /verify
+
+# Specify release variables
+ARG RLS_VERSION=1.14.4
+ARG RLS_OS=linux
+ARG RLS_LIB=gnu
+ARG RLS_ARCH=x86_64
+
+# Automatically detect architecture
+RUN set -ex && ARCHITECTURE=$(dpkg --print-architecture) && \
+    if [ "${ARCHITECTURE}" = "amd64" ]; then RLS_ARCH=x86_64 ; fi \
+    && if [ "${ARCHITECTURE}" = "arm64" ]; then RLS_ARCH=aarch64; fi \
+    && if [ "${ARCHITECTURE}" = "armhf" ]; then RLS_ARCH=arm && RLS_LIB=gnuabihf; fi \
+    && if [ "${ARCHITECTURE}" = "i386" ]; then RLS_ARCH=i686-pc; fi
+
+ARG RLS_FILE_NAME=dogecoin-${RLS_VERSION}-${RLS_ARCH}-${RLS_OS}-${RLS_LIB}.tar.gz
```

Comment:
Will review deeper tomorrow, but you won't get the expected behavior for cross architecture. RLS_ARCH value is fixed.

```Dockerfile
ARG RLS_ARCH=420

RUN RLS_ARCH=69

RUN echo ARCH : $RLS_ARCH
#output 420
```

Reply:
I see the issue, thanks! Fixing.

Reply:
Fixed by constructing and persisting the release filename into a temporary file, then re-assigning that when we need it.